src/Nethermind/Nethermind.State/PersistentStorageProvider.cs

```diff
@@ -428,7 +428,14 @@ public override void ClearStorage(Address address)
             base.ClearStorage(address);
 
             // Bit heavy-handed, but we need to clear all the cache for that address
-            _blockCache.Remove(address);
+            if (_blockCache.TryGetValue(address, out Dictionary<UInt256, byte[]> values))
+            {
+                foreach (UInt256 storageSlot in values.Keys)
+                {
+                    ref var value = ref CollectionsMarshal.GetValueRefOrNullRef(values, storageSlot);
```

Comment:
Why `CollectionsMarshal` and not just `values[storageSlot] = StorageTree.EmptyBytes;`?